### DISCUSSION TITLE
Can't Import Pictures via Crontab

### DISCUSSION THREAD

**Discussion by TechD123** [photoprism/photoprism#3118] [Q&A] [answered]
Despite my limited knowledge of docker, I finally realized how I can trigger the import from a specific folder with a command. I've edited the docker-compose.yml file to mount the folder in question (it is filled via syncthing). This is the command:
`docker-compose exec photoprism photoprism import /photoprism/import/syncthing`

This command works perfectly when I run it as root and so I added it to crontab (i.e. `sudo crontab -e`, add `00 * * * *` to execute every hour, followed by above command).

Via the syslog (`grep photoprism /var/log/syslog`) I can see that the command is executed every hour, for instance:
`Jan 17 12:00:02 rpi4 CRON[31701]: (root) CMD (docker-compose exec photoprism photoprism import /photoprism/import/syncthing)`

But the pictures aren't imported when the command is run via crontab – they remain in the `syncthing` folder and don't show up in PhotoPrism :(
What am I missing here? Any insights appreciated!

**Comment by beefybytes** [comment, marked as answer]
I ran into a similar issue and had to add `-T` argument to `docker-compose exec` inside my cron script to disable pseudo-TTY allocation:
`docker-compose exec -T photoprism photoprism import /photoprism/import/syncthing`

**Reply by TechD123** [reply]
Perfect, that makes sense. Thank you! :)

**Comment by dvpfig** [comment]
If you are running PhotoPrism with docker-compose, I believe it's simpler (and recommended) to use Ofelia to trigger the regular jobs to import/backup/...

The example here has an example: 
https://github.com/photoprism/photoprism/blob/develop/setup/docker/cloud/docker-compose.yml

Add this to your docker-compose:
```yaml
  ofelia:
    restart: always
    image: mcuadros/ofelia:latest
    container_name: ofelia
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "./jobs.ini:/etc/ofelia/config.ini"
```  

And then edit a jobs.ini as the example: 
https://github.com/photoprism/photoprism/blob/develop/setup/docker/cloud/jobs.ini

```ini
[job-exec "photoprism convert"]
schedule = @daily
container = photoprism
command =  photoprism convert
no-overlap = true

[job-exec "photoprism backup"]
schedule = @daily
container = photoprism
command =  photoprism backup -a -i -f /photoprism/storage/index-backup.sql
no-overlap = true
```  

For your case, you can add this for example: 
```ini
[job-exec "photoprism import"]
schedule = @hourly
container = photoprism
command =  photoprism import /photoprism/import/syncthing
no-overlap = true
```

**Reply by TechD123** [reply]
I did try ofelia before, but gave up on it after it didn't work – turns out I didn't specify the import path, as I assumed from the official example it wasn't needed :P

This was what I had in my config.ini:

```
[job-exec "photoprism import"]
schedule = @every 2h
container = photoprism
command =  photoprism import [naively forgetting to add the path]
no-overlap = true
```

I'll give it another shot now, thank you for clarifying!

**Reply by lastzero** [reply]
Feel free to contribute and improve the official example! ❤️

**Reply by TechD123** [reply]
Ah, the beauty of open source! Done, hope I did it right (my first contribution to a project).

**Comment by mrpeu** [comment]
Does the new variable [`PHOTOPRISM_INDEX_SCHEDULE`](https://docs.photoprism.app/getting-started/config-options/#indexing) do the same as this ofelia example?

**Reply by lastzero** [reply]
Even better 🌈

**Reply by mrpeu** [reply]
Hmm, I've yet to succeed. Neither with the syntax hinted at in the doc ("@every 15min") nor a more common CRON syntax seems to work. I also have a syncthing'd folder that I'd like Photoprism to stay in touch with. Being on a NAS, maybe I should look into the WebDAV stuff.

**Reply by lastzero** [reply]
Did you verify the config with the `photoprism config` command and watch the logs in debug or trace mode? You could also reduce the time to every two minutes for testing e.g. via `*/2 * * * *`.

**Reply by lastzero** [reply]
Is it possible that you try to IMPORT, rather then INDEX? The INDEX schedule does not IMPORT.

**Reply by mrpeu** [reply]
I would prefer INDEX indeed. When I click Library>Index>start, the new pictures do appear shortly in the gallery (search). The logs do show activity every 2 minutes but no obvious error.
I'll try to ask for help in the Gitter or a new github discussion to not bother you here.
